### ISSUE TITLE
Existing books read time gets set to <1 hour when new book added to series (docker only)

### ISSUE THREAD

**Issue by scare376** [Kareadita/Kavita#1739]
**Describe the bug**
When a new book is added to a series books with > 1 hour read time gets set to <1 hour

**To Reproduce**
1. Add a series
2. Books read time will be correct.
3. Add a new book to the series. 
4. Read time of existing books in series will be set to <1 hour. (Page needs to be refreshed to see the change)

**Expected behavior**
Read time is unchanged when adding a new book.

**Screenshots**
![image](https://user-images.githubusercontent.com/2068470/211682009-3af880b8-f6bb-43bb-9354-9db05249b1f6.png)
![image](https://user-images.githubusercontent.com/2068470/211682096-0327d401-4fd4-4e7d-ac47-0a8b0be7f043.png)

**Desktop (please complete the following information):**
 - OS: Docker
 - Browser: Opera GX
 - Version: 0.6.1.26

**Additional context**

**Comment by majora2007** [comment]
I'm not able to reproduce this on my dev machine, nor via Unit Tests. New chapters added to an existing series are correctly adding for read time and word counts.

**Comment by scare376** [comment]
Did you test with epubs at all?

**Comment by majora2007** [comment]
Yes, I tested against a Book library with Epubs. I did not test with archives.

**Comment by scare376** [comment]
I dont know what to tell you. I just reproduced it in a fresh stable and fresh nightly instance.

**Comment by majora2007** [comment]
Hmm...I seriously can't reproduce. In your example, the first volume BEFORE you added the new volume, what was the read time?

**Comment by majora2007** [comment]
Okay I tried again (via UI in case it's a UI issue) and it's working. I validated again through the UI and my existing series aren't having any change in their reading time estimate. 

@therobbiedavis or @ThePromidius can you also test this out to see if you can reproduce?

**Comment by scare376** [comment]
Ok I tested it again. This time baremetal in windows. It did not change when i added a new volume initially. but i hit scan series after (without changing any files) and it changed the read time. the read time is 1-4 hours like it shows in the image but ive had it happen on longer series as well. particularly one that said roughly 2-5 hours per volume that i added the other day.

**Comment by majora2007** [comment]
When you first add the volume, a task is scheduled to run which does the calculation, so when you hit scan, it probably just sent the update to the UI to show what the calculation did.

**Comment by scare376** [comment]
https://user-images.githubusercontent.com/2068470/215702954-e03e5bbc-154e-48e2-ba1d-ac89535c46c3.mp4

**Comment by scare376** [comment]
This happens on all my books. The only time it did something different was on baremetal where I added a new volume. Both showed an estimated read time. I scanned again, refreshed and it went back to <1 hour

**Comment by zeedif** [comment]
I didn't know that the <1 was a problem, in fact in some of my epubs it also shows up like this.
![image](https://user-images.githubusercontent.com/60901341/216447434-b430116f-deca-44f4-9486-2ed6d05c8803.png)

**Comment by majora2007** [comment]
I haven't given up on this yet. I double checked the logic and it's working fine. I checked the UI code and it only shows <1 when min or max hours to read is 0. 

My only thinking is the estimation code (happens in word count stage) gets skipped for some issues. I'm having a hard time reproducing though.

**Comment by bnigh** [comment]
I am facing a similar issue:
- The read times for some books in a series are incorrect (but the word counts are correct)
- The read times and word counts for the series (overview) are incorrect

---

I attempted to triage this issue

TLDR - I believe the issue is due to skipping processing chapters if they haven't been updated:
- https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/API/Services/Tasks/Metadata/WordCountAnalyzerService.cs#L159-L161

---

**Some triaging notes**

Looking at api invocations, I notice, that the backend is sending the read times (min, max, avg) as 0s.

Reading through the code, it seems that each book in a series is represented by a volume with exactly 1 chapter.

From the ui for the individual books (volumes)
- the word counts are fetched from the chapter data (which is correct)
  - https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/UI/Web/src/app/cards/entity-info-cards/entity-info-cards.component.ts#L93-L96
- the read time data are fetched from the volume data (which is incorrectly set to 0)
  - https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/UI/Web/src/app/cards/entity-info-cards/entity-info-cards.component.ts#L104-L109




Looking through the api code, I believe the bug occurs when processing series
- https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/API/Services/Tasks/Metadata/WordCountAnalyzerService.cs#L146

This function updates the wordCounts for the series and volumes to `0`
- https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/API/Services/Tasks/Metadata/WordCountAnalyzerService.cs#L150-L153

The counts are then updated later for each chapter / volume
 - https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/API/Services/Tasks/Metadata/WordCountAnalyzerService.cs#L202-L204

However, chapters that already exist and are not updated skip the processing
- https://github.com/Kareadita/Kavita/blob/baf7c9eb926c2c7d9c61b68f113d47c11e9fadd7/API/Services/Tasks/Metadata/WordCountAnalyzerService.cs#L159-L161

These skipped chapters would then be excluded from the series and volume word counts and read times

**Comment by majora2007** [comment]
I think you might be right @bnigh in that some chapters are being skipped. However, this is a normal part of Kavita and applies to other areas. I did notice that after a Scan Series, we weren't fully recounting words, which it should have, so I've fixed that up. That will bypass the HasFileChangedSinceLastScan completely, so users don't need to do the manual Analyze Files task.

**Comment by majora2007** [comment]
This will be released in v0.7.4. I will close the issue as I believe that fixes, but we can reopen it if it does not.
